source/fundamentals/builders/indexes.txt

```diff
@@ -11,3 +11,181 @@ Indexes Builders
    :class: singlecol
 
 .. _indexes-builders:
+
+Overview 
+--------
+
+This guide shows you how to specify **indexes** using the Java Driver
+:doc:`builders </fundamentals/builders/>`. Index builders provide
+helper methods for the following types of indexes:
+
+- :ref:`Ascending <index-ascending>`
+- :ref:`Descending <index-descending>`
+- :ref:`Compound Indexes <index-compound-indexes>`
+- :ref:`Text Indexes <index-text-indexes>`
+- :ref:`Hashed Indexes <index-hashed-indexes>`
+- :ref:`Geospatial Indexes <index-geospatial-indexes>`
+
+Indexes store a subset of the collection’s data set. The index stores
+the value of a specific field or set of fields, ordered by the value of
+the field. See our guide on :doc:`Indexes </fundamentals/indexes>` to
+see examples of queries covered by indexes. 
+
+The :java-core-api:`Indexes <com/mongodb/client/model/Indexes.html>`
+class provides static factory methods for all the MongoDB index types.
+Each method returns a :java-docs:`Bson <apidocs/bson/org/bson/conversions/Bson.html?is-external=true>` 
+instance, which you can pass to
+:java-sync-api:`createIndex() <com/mongodb/client/MongoCollection.html#createIndex(org.bson.conversions.Bson)>`.
+
+.. tip::
+
+   For brevity, you may choose to import all methods of the
+   :java-core-api:`Indexes <com/mongodb/client/model/Indexes.html>`
+   class statically: 
+
+   .. code-block:: java
+
+      import static com.mongodb.client.model.Indexes.*;
+
+   The examples below assume this static import.
+
+.. _index-ascending:
+
+Ascending
+---------
+
+An ascending index enables you to sort query results by the value of the
+indexed field(s) from smallest to largest.
+
+In order to create an ascending index, first call the
+:java-core-api:`ascending() <com/mongodb/client/model/Indexes.html#ascending(java.lang.String...)>`
+builder method to create a ``Bson`` instance that represents the index
+document, passing the name or names of the fields you want to index.
+Then, call the ``createIndex()`` method on the collection, passing the ``Bson``
+instance that contains the index document.
+
+.. note:: 
+
+   If an ascending or a descending index is on a single field, the sort
+   operation on the field can be in either direction. 
```

Comment:
Nitpick: I think this might be a little more clear in describing the sort behavior.

```suggestion
   If you have an ascending or a descending index on a single field, MongoDB
   can sort using the index in either direction. 
```